File: algorithms/bit-manipulation/bit-reversal/csharp/BitReversal.cs
public class BitReversal
{
    public static long Reverse(long n)
    {
        uint val = (uint)n;
        uint result = 0;
        for (int i = 0; i < 32; i++)
        {
            result = (result << 1) | (val & 1);
            val >>= 1;
        }
        return (long)result;
    }
}

File: algorithms/bit-manipulation/bit-reversal/rust/bit_reversal.rs
pub fn bit_reversal(n: i64) -> i64 {
    let mut val = n as u32;
    let mut result: u32 = 0;
    for _ in 0..32 {
        result = (result << 1) | (val & 1);
        val >>= 1;
    }
    result as i64
}
